compiler/src/dotty/tools/dotc/sbt/ExtractAPI.scala

```diff
@@ -62,13 +65,59 @@ class ExtractAPI extends Phase {
   // definitions, and `PostTyper` does not change definitions).
   override def runsAfter: Set[String] = Set(transform.PostTyper.name)
 
+  override def runOn(units: List[CompilationUnit])(using Context): List[CompilationUnit] =
+    val nonLocalClassSymbols = new mutable.HashSet[Symbol]
+    val ctx0 = ctx.withProperty(NonLocalClassSymbolsInCurrentUnits, Some(nonLocalClassSymbols))
+    val units0 = super.runOn(units)(using ctx0)
+    ctx.withIncCallback(recordNonLocalClasses(nonLocalClassSymbols, _))
+    units0
+  end runOn
+
+  private def recordNonLocalClasses(nonLocalClassSymbols: mutable.HashSet[Symbol], cb: interfaces.IncrementalCallback)(using Context): Unit =
+    for cls <- nonLocalClassSymbols if !cls.isLocal do
```

Comment:
Why is the set called *nonLocal* symbols but you still need to do `if !cls.isLocal`?

Reply:
ah yes this was a holdover from https://github.com/sbt/zinc/commit/856d4162127927cb9a6c37a1649cc42d1871a815 which also does this seemingly redundant check - I never tested removing it

Reply:
one thing it is actually used for in sbt is deciding if pipelining should begin (it checks the APIs for classes that define macros, and skips pipelining if true, these API would not be available without the change)

Reply:
moved the check to where we actually add the class